src/Security/LdapUserProvider.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+/*
+ * This file is part of the package t3g/symfony-ldap-bundle.
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace T3G\Bundle\LdapBundle\Security;
+
+use Symfony\Component\Ldap\Entry;
+use Symfony\Component\Ldap\Exception\ConnectionException;
+use Symfony\Component\Ldap\LdapInterface;
+use Symfony\Component\Security\Core\Exception\InvalidArgumentException;
+use Symfony\Component\Security\Core\Exception\UnsupportedUserException;
+use Symfony\Component\Security\Core\Exception\UsernameNotFoundException;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Symfony\Component\Security\Core\User\UserProviderInterface;
+use T3G\Bundle\LdapBundle\Entity\LdapUser;
+
+class LdapUserProvider implements UserProviderInterface
+{
+    /**
+     * @var LdapInterface
+     */
+    private $ldap;
+    /**
+     * @var string
+     */
+    private $baseDn;
+    /**
+     * @var string
+     */
+    private $searchDn;
+    /**
+     * @var string
+     */
+    private $searchPassword;
+    /**
+     * @var array
+     */
+    private $defaultRoles;
+    /**
+     * @var string
+     */
+    private $defaultSearch;
+    /**
+     * @var string
+     */
+    private $userClass;
+    /**
+     * @var array
+     */
+    private $roleMapping;
+
+    /**
+     * @param LdapInterface $ldap
+     * @param string $baseDn
+     * @param string $searchDn
+     * @param string $searchPassword
+     * @param array $defaultRoles
+     * @param array $roleMapping
+     * @param string $userClass
+     */
+    public function __construct(
+        LdapInterface $ldap,
+        string $basedn,
+        string $searchDn,
+        string $searchPassword,
+        array $defaultRoles,
+        array $roleMapping,
+        string $userClass
+    ) {
+        $this->ldap = $ldap;
+        $this->baseDn = $basedn;
+        $this->searchDn = $searchDn;
+        $this->searchPassword = $searchPassword;
+        $this->defaultRoles = $defaultRoles;
+        $this->defaultSearch = '(uid={username})';
+        $this->roleMapping = $roleMapping;
+        $this->userClass = $userClass;
+    }
+
+    /**
+     * Creates the user object, assigns roles from isMemberOf attribute
+     * and sets display name from LDAP attribute.
+     *
+     * @param string $username
+     * @param Entry $entry
+     * @return UserInterface
+     */
+    protected function loadUser($username, Entry $entry): UserInterface
+    {
+        $displayName = null;
+        $uid = $this->getAttributeValue($entry, 'uid');
+        if ($entry->hasAttribute('displayName')) {
+            $displayName = $this->getAttributeValue($entry, 'displayName');
+        }
+        if (!$entry->hasAttribute('isMemberOf')) {
+            // If user does not have this attribute at all, he's just ROLE_USER
+            return new User($username, null, $displayName, $this->defaultRoles);
```

Comment:
Class `T3G\Bundle\LdapBundle\Security\User` does not exist in the package. Is this a missing dependency?